Fast Pair documentation alignment with the recent changes to sysbuild Kconfig options (#20900)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_3.0.rst

This section describes the changes related to libraries.

Google Fast Pair
----------------

.. toggle::

   For applications and samples using the :ref:`bt_fast_pair_readme` library:

>    * If you use the sysbuild feature for generating a hex file with the Fast Pair provisioning data, you must align your application with the newest approach for passing the provisioning parameters (the Model ID and the Anti-Spoofing Private Key).

Reviewer: ```suggestion
   * If you use the sysbuild for generating a hex file with the Fast Pair provisioning data, you must align your application with the new approach for passing the provisioning parameters (the Model ID and the Anti-Spoofing Private Key).
```

     The ``FP_MODEL_ID`` and ``FP_ANTI_SPOOFING_KEY`` CMake variables are replaced by the corresponding ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig options that are defined at the sysbuild level.

>      The additional build parameters for Fast Pair are no longer valid:

Reviewer 2: ```suggestion
     The following additional build parameters for Fast Pair are no longer valid:
```

     ``-DFP_MODEL_ID=0xFFFFFF -DFP_ANTI_SPOOFING_KEY=AbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbA=``

>      And must be replaced with the following parameters:

Reviewer: ```suggestion
     And can for example be replaced with the following parameters:
```
We could set sysbuild Kconfigs also in another way

This section describes the changes related to libraries.

Google Fast Pair
----------------

.. toggle::

   For applications and samples using the :ref:`bt_fast_pair_readme` library:

   * If you use sysbuild for generating a hex file with the Fast Pair provisioning data, you must align your application with the new approach for passing the provisioning parameters (the Model ID and the Anti-Spoofing Private Key).
     The ``FP_MODEL_ID`` and ``FP_ANTI_SPOOFING_KEY`` CMake variables are replaced by the corresponding ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig options that are defined at the sysbuild level.
     The following additional build parameters for Fast Pair are no longer valid:

     ``-DFP_MODEL_ID=0xFFFFFF -DFP_ANTI_SPOOFING_KEY=AbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbA=``

     And must be replaced with the new sysbuild Kconfig options.

>      You can provide them as additional build parameters to the build command:

Reviewer 2: ```suggestion
     You can provide them as additional build parameters to the build command as follows:
```

     ``-DSB_CONFIG_BT_FAST_PAIR_MODEL_ID=0xFFFFFF -DSB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY=\"AbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbA=\"``

>      This exemplary method can be replaced with any other configuration method that is supported by sysbuild.

Reviewer 2: ```suggestion
     You can replace this exemplary method with any other configuration method that is supported by sysbuild.
```

     .. note::

>         To avoid build failures, you must surround the string value for the Anti-Spoofing Private Key parameter with the ``\"`` character instead of the typical ``"`` character.

Reviewer 2: ```suggestion
        To avoid build failures, you must surround the string value for the Anti-Spoofing Private Key parameter with the ``\"`` characters instead of the typical ``"`` character.
```

>         To avoid build failures, you must surround the string value for the Anti-Spoofing Private Key parameter with the ``\"`` characters instead of the typical ``"`` character.

Reviewer 3: this might work for linux but probably doesn't work for windows or vice-versa

Author: thanks, tested using Windows and found the syntax that works for Windows

>         This backlash character is only required when you pass the ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig option as an additional build parameter.

Reviewer 2: ```suggestion
        The backlash character is only required when you pass the ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig option as an additional build parameter.
```

This section describes the changes related to libraries.

Google Fast Pair
----------------

.. toggle::

   For applications and samples using the :ref:`bt_fast_pair_readme` library:

   * If you use sysbuild for generating a hex file with the Fast Pair provisioning data, you must align your application with the new approach for passing the provisioning parameters (the Model ID and the Anti-Spoofing Private Key).
     The ``FP_MODEL_ID`` and ``FP_ANTI_SPOOFING_KEY`` CMake variables are replaced by the corresponding ``SB_CONFIG_BT_FAST_PAIR_MODEL_ID`` and ``SB_CONFIG_BT_FAST_PAIR_ANTI_SPOOFING_PRIVATE_KEY`` Kconfig options that are defined at the sysbuild level.
     The following additional build parameters for Fast Pair are no longer valid:

     ``-DFP_MODEL_ID=0xFFFFFF -DFP_ANTI_SPOOFING_KEY=AbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbAbA=``

>      And must be replaced with the new sysbuild Kconfig options.

Reviewer 2: ```suggestion
     You must replace them with the new sysbuild Kconfig options.
```

Author: fixed
